### ISSUE TITLE
unspecified-encoding error when the encoding is specified with positional argument

### ISSUE THREAD

**Issue by eight04** [PyCQA/pylint#5638]
### Bug description

*test.py*
```py
from pathlib import Path
Path("test.txt").write_text("中文")
Path("test2.txt").write_text("中文", "utf8")
Path("test3.txt").write_text("中文", encoding="utf8")
```
Get an error on L3 even the encoding is already specified. I think this also applies to `read_text()`.

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:2:0: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
test.py:3:0: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
```

### Expected behavior

L3 should not raise `unspecified-encoding`.

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.2
Python 3.8.1 (tags/v3.8.1:1b293b6, Dec 18 2019, 22:39:24) [MSC v.1916 32 bit (Intel)]
```

### OS / Environment

Windows 10 64-bit / CMD
